trunk/SUAVE/Components/Energy/Converters/Compression_Nozzle.py

```diff
@@ -97,42 +99,69 @@ def compute(self,conditions):
         self.
           pressure_ratio                      [-]
           polytropic_efficiency               [-]
-        """           
-        #unpack the values
-        
+        """
+
         #unpack from conditions
         gamma   = conditions.freestream.isentropic_expansion_factor
         Cp      = conditions.freestream.specific_heat_at_constant_pressure
         Po      = conditions.freestream.pressure
-        R       = conditions.freestream.universal_gas_constant
-        
+        Mo      = conditions.freestream.mach_number
+        R       = conditions.freestream.gas_specific_constant
+
         #unpack from inpust
         Tt_in   = self.inputs.stagnation_temperature
         Pt_in   = self.inputs.stagnation_pressure
-        
+
         #unpack from self
-        pid     =  self.pressure_ratio
-        etapold =  self.polytropic_efficiency
-        
+        pid                     =  self.pressure_ratio
+        etapold                 =  self.polytropic_efficiency
+        compressibility_effects =  self.compressibility_effects
+
         #Method to compute the output variables
-        
+
         #--Getting the output stagnation quantities
-        Pt_out  = Pt_in*pid
         Tt_out  = Tt_in*pid**((gamma-1)/(gamma*etapold))
         ht_out  = Cp*Tt_out
-        
-        # in case pressures go too low
-        if np.any(Pt_out<Po):
-            warn('Pt_out goes too low',RuntimeWarning)
-            Pt_out[Pt_out<Po] = Po[Pt_out<Po]
-        
-        
-        #compute the output Mach number, static quantities and the output velocity
-        Mach    = np.sqrt( (((Pt_out/Po)**((gamma-1.)/gamma))-1.) *2./(gamma-1.) )
-        T_out   = Tt_out/(1+(gamma-1)/2*Mach*Mach)
+
+        if compressibility_effects :
+
+            # Checking from Mach numbers below, above 1.0
+            i_low  = Mo <= 1.0
+            i_high = Mo > 1.0
+
+            #initializing the arrays
```

Comment:
Let's initialize arrays a better way than this. This is slow.

Reply:
Not sure best way.